lib/lita/handlers/jls.rb

```diff
@@ -137,6 +143,72 @@ def cla(msg)
         push_exception(e, :project => "#{user}/#{project}", :pr => pr)
       end
 
+      def migrate_pr(msg)
+        source_url = msg.match_data["source_url"]
+        destination_url = msg.match_data["destination_url"]
+        if source_url.nil? || destination_url.nil?
+          raise "Invalid paramaters provided #{msg}"
+        end
+
+        pr_num = source_url.strip.split("/").last
+        if pr_num.nil? || pr_num !~ /\d+/
+          raise "Invalid pull request provided #{source_url}"
+        end
+
+        destination_github_parser = parse_github_url(destination_url)
+        source_github_parser = parse_github_url(source_url)
+
+        source_github_pr = github_get_pr("#{source_github_parser.user}/#{source_github_parser.project}", pr_num)
+
+        # Clone destination dir, patch and then push branch
+        repository = LitaJLS::Repository.new(destination_github_parser)
+        repository.clone if Dir["#{repository.git_path}/*"].empty?
+        repository.switch_branch("master")
+
+        # create a branch like pr/1234
+        pr_branch = "pr/#{pr_num}"
+        repository.delete_local_branch(pr_branch, true)
+        repository.switch_branch(pr_branch, true)
+
+        response = get_patch(source_github_pr[:patch_url])
+        patch_file = Tempfile.new("#{pr_num}.patch")
+
+        # Apply patch on repo
+        begin
+          #TODO: Use chunked writes
+          patch_file.write(response.body)
+          patch_file.close
+          repository.git_patch(patch_file.path)
+        rescue => e
+          msg.reply("Error while migrating pr: #{e}")
+          push_exception(e, :source_url => source_url,
+                         :destination_url => destination_url)
+        ensure
+          patch_file.unlink
+        end
+
+        # create the migrated PR in the destination repo
+        github_create_pr("#{destination_github_parser.user}/#{destination_github_parser.project}",
+                         pr_branch, source_github_pr[:title], source_github_pr[:body])
+      end
```

Comment:
I would move the `patch_file.write` and `patch_file.close` into the `get_patch` method. 

Maybe we could rename this method to 'download_file' and make it return a `File` instance, this would make the call reusable?